subsys/shell/shell_uart.c

```diff
@@ -282,7 +288,9 @@ static int enable_shell_uart(struct device *arg)
 
 	return 0;
 }
-SYS_INIT(enable_shell_uart, POST_KERNEL, 0);
+SYS_INIT(enable_shell_uart, POST_KERNEL,
```

Comment:
This might not be right.  If `CONFIG_UART_SHELL_ON_DEV_NAME` and `CONFIG_UART_CONSOLE_ON_DEV_NAME` are both true then the shell would be enabled on the device before the console is enabled (in level `APPLICATION` at priority `CONFIG_UART_CONSOLE_INIT_PRIORITY`).